static/app/utils/replays/hooks/useReplayData.tsx

```diff
@@ -233,4 +242,38 @@ function useReplayData({
   };
 }
 
+// see https://github.com/getsentry/sentry/pull/47859
+// replays can apply to many projects when incorporating backend errors
+// this makes having project in the `replaySlug` obsolete
+// we must keep this url schema for now for backward compat but we should remove it at some point
+// TODO: remove support for projectSlug in replay url?
+function parseReplaySlug(replaySlug: string) {
+  const maybeProjectSlugAndReplayId = replaySlug.split(':');
+  if (maybeProjectSlugAndReplayId.length === 2) {
+    const [projectSlug, replayId] = maybeProjectSlugAndReplayId;
+    return {
+      projectSlug,
+      replayId,
+    };
+  }
+
+  // if there is no projectSlug then we assume we just have the replayId
+  // all other cases would be a malformed url
+  return {
+    projectSlug: null,
+    replayId: maybeProjectSlugAndReplayId[0],
+  };
+}
+
+function makeFetchReplayApiUrl(
+  orgSlug: string,
+  projectSlug: string | null,
+  replayId: string
+) {
+  if (projectSlug) {
+    return `/projects/${orgSlug}/${projectSlug}/replays/${replayId}/`;
+  }
+  return `/organizations/${orgSlug}/replays/${replayId}/`;
+}
+
```

Comment:
note to self; reviewing this after the fact... we only need to support the replayId codepath and can ignore projectslug completely